exercises/08-exercise.py

```diff
@@ -31,26 +33,69 @@ def __init__(self, resource_path, chunk_size=1024):
             user specifices size of chunk for data, default is 1024
         '''
         self.chunks_of_data = []
-        for chunk in pd.read_csv(resource_path, chunksize=chunk_size, 
-                names =['time (s)', 'time (ns)', 'index', 'counts'], 
-                delimiter = " ", header=None):
-            chunk['adc'] = chunk['counts'].apply(adc2counts)
-            chunk['timestamp'] = chunk['time (s)'] + 1e-9*chunk['time (ns)']
-            chunk = chunk.drop(columns = ['time (s)', 'time (ns)', 'index', 'counts'])
-            chunk = chunk[['timestamp','adc']]
-            self.chunks_of_data.append(chunk)
-
-    def __call__(self, chunk_num):
+        chunk = [data for data in pd.read_csv(resource_path, 
+            chunksize=chunk_size, delimiter = " ", header = None) ]
+        
+        _, num_cols = chunk[0].shape
+
+        if(num_cols == 5):
```

Comment:
nice! there are ways to simplify this and generalize this. one could also try
```py3
columns = ['time_s', 'time_ns', 'index']
columns_left = num_cols - len(columns)
columns  = columns + [f'adc {i}' for i in range(columns_left)]
```
to name them `adc 0` , `adc 1` etc.
continuing with this, you could simplify the code a bit. it would also work for more than 2 extra columns